app_capabilities.go

```diff
@@ -78,6 +78,26 @@ func (e AppCapabilities) Create(stackId, appId, envId int64, capabilities []*typ
 	return createdCaps, nil
 }
 
+func (e AppCapabilities) Replace(stackId, appId, envId int64, capabilities []*types.Capability, blocks []*types.Block) ([]*types.Capability, error) {
+	input := CreateCapabilitiesInput{
+		Capabilities: capabilities,
+		Blocks:       blocks,
+	}
+	rawPayload, _ := json.Marshal(input)
+	res, err := e.Client.Do(http.MethodPut, e.basePath(stackId, appId, envId), nil, nil, json.RawMessage(rawPayload))
+	if err != nil {
+		return nil, err
+	}
+
+	var caps []*types.Capability
```

Comment:
You can replace this rest of this function with...
```suggestion
	return response.ReadJsonVal[[]*types.Capability](res)
```